iceoryx_posh/include/iceoryx_posh/internal/roudi/roudi_multi_process.hpp

```diff
@@ -41,17 +41,33 @@ using namespace iox::units::duration_literals;
 class RouDiMultiProcess
 {
   public:
+
+    // indicate whether the message queue thread will start directly or deferred
+    // this is important for derived classes which may need to initialize their members before the thread starts
+    enum class MQThreadMode {
+      START,
+      DEFER_START
```

Comment:
This option seems redundant since the deferred start is not managed by this class. Could we get away with just the one option?

I would also suggest a slightly more  descriptive name like:
```
    enum class MQThreadMode {
        IMMEDIATE_START
    };
```